### PR TITLE
chore(forms): update stories for forms

### PR DESCRIPTION
**Pull request by andyfeds**
- CDE-1529 [Storybook] - Refactor forms stories to CSFv3

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: [CDE-1529](https://jira.eng.vmware.com/browse/CDE-1529)

## What is the new behavior?
Refactored form stories to CSFv3

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .storybook/stories/forms/forms-input-states.stories.ts [hunk 1]
```diff
@@ -159,6 +162,21 @@ const defaultParameters: Parameters = {
...
+
+export const disabledStates: StoryObj = {
+  render: FormInputTemplate,
+  args: { isDisabled: true },
+};
+
+export const errorStates: StoryObj = {
+  render: FormInputTemplate,
+  args: { isError: true },
+};
```
Comment (dtsanevmw) [inline .storybook/stories/forms/forms-input-states.stories.ts:177]:
```suggestion
export const DisabledStates: StoryObj = {
  render: FormInputTemplate,
  args: { isDisabled: true },
};

export const ErrorStates: StoryObj = {
  render: FormInputTemplate,
  args: { isError: true },
};
```

File: .storybook/stories/forms/forms-reactive.stories.ts [hunk 2]
```diff
@@ -119,3 +107,17 @@ function getForm() {
...
+
+export const verticalLayout: StoryObj = {
+  render: ReactiveFormTemplate,
+  args: { namePlaceholder: 'Test placeholder', clrLayout: ClrFormLayout.VERTICAL },
+};
+
+export const compactLayout: StoryObj = {
+  render: ReactiveFormTemplate,
+  args: { namePlaceholder: 'Test placeholder', clrLayout: ClrFormLayout.COMPACT },
+};
```
Comment (dtsanevmw) [inline .storybook/stories/forms/forms-reactive.stories.ts:123]:
```suggestion
export const VerticalLayout: StoryObj = {
  render: ReactiveFormTemplate,
  args: { namePlaceholder: 'Test placeholder', clrLayout: ClrFormLayout.VERTICAL },
};

export const CompactLayout: StoryObj = {
  render: ReactiveFormTemplate,
  args: { namePlaceholder: 'Test placeholder', clrLayout: ClrFormLayout.COMPACT },
};
```

File: .storybook/stories/forms/forms-required-mark.stories.ts [hunk 3]
```diff
@@ -158,3 +146,17 @@ function getForm() {
...
+
+export const verticalLayout: StoryObj = {
+  render: RequiredMarkTemplate,
+  args: { namePlaceholder: 'Test placeholder', clrLayout: ClrFormLayout.VERTICAL },
+};
+
+export const compactLayout: StoryObj = {
+  render: RequiredMarkTemplate,
+  args: { namePlaceholder: 'Test placeholder', clrLayout: ClrFormLayout.COMPACT },
+};
```
Comment (dtsanevmw) [inline .storybook/stories/forms/forms-required-mark.stories.ts:162]:
```suggestion
export const VerticalLayout: StoryObj = {
  render: RequiredMarkTemplate,
  args: { namePlaceholder: 'Test placeholder', clrLayout: ClrFormLayout.VERTICAL },
};

export const CompactLayout: StoryObj = {
  render: RequiredMarkTemplate,
  args: { namePlaceholder: 'Test placeholder', clrLayout: ClrFormLayout.COMPACT },
};
```

File: .storybook/stories/forms/forms-template-driven.stories.ts [hunk 4]
```diff
@@ -101,3 +100,17 @@ function getForm() {
...
+
+export const verticalLayout: StoryObj = {
+  render: TemplateDrivenStory,
+  args: { namePlaceholder: 'Test placeholder', clrLayout: ClrFormLayout.VERTICAL },
+};
+
+export const compactLayout: StoryObj = {
+  render: TemplateDrivenStory,
+  args: { namePlaceholder: 'Test placeholder', clrLayout: ClrFormLayout.COMPACT },
+};
```
Comment (dtsanevmw) [inline .storybook/stories/forms/forms-template-driven.stories.ts:116]:
```suggestion
export const VerticalLayout: StoryObj = {
  render: TemplateDrivenStory,
  args: { namePlaceholder: 'Test placeholder', clrLayout: ClrFormLayout.VERTICAL },
};

export const CompactLayout: StoryObj = {
  render: TemplateDrivenStory,
  args: { namePlaceholder: 'Test placeholder', clrLayout: ClrFormLayout.COMPACT },
};
```

File: .storybook/stories/forms/forms-input-states.stories.ts (hunk 1 above)
Comment (andyfeds) [reply .storybook/stories/forms/forms-input-states.stories.ts]: Changed the story names to Uppercase.

**Review by dtsanevmw — APPROVED** [review]

**Pull request merged**